Mono.Addins/Mono.Addins/ExtensionContext.cs

```diff
@@ -1087,16 +1088,17 @@ ExtensionLoadData GetAddinExtensions (string id, ExtensionPoint ep)
 			}
 			if (!pinfo.Enabled || pinfo.Version != Addin.GetIdVersion (id))
 				return null;
-				
+
 			// Loads extensions defined in each module
-			
+			var name = Addin.GetIdName (id);
+
 			ExtensionLoadData data = null;
 			AddinDescription conf = pinfo.Description;
-			GetAddinExtensions (conf.MainModule, id, ep, ref data);
+			GetAddinExtensions (conf.MainModule, name, ep, ref data);
```

Comment:
I don't like this change, it may have unwanted side effects.

Reply:
The main problem with specific version checking is that:
MD.Ide 7.5 references MD.Core 7.5
MD.AddinMaker depends on MD.Ide 7.0, MD.Core 7.0 which resolves to MD.Ide.7.5 and MD.Core 7.5

If we change this, we won't have any side-effects outside of addin dependency tree.

See the line below, it was _already_ doing non-version checked. Unless we have some way of mapping 7.0 to 7.5 in all the caches, we'll end up having bad data in the cache.


https://github.com/mono/mono-addins/pull/97/files#diff-e867160bdb98148a444c4bd3453b177dR987

When I made it write the full version, MD.AddinMaker referenced these in the cache:
```
MD.Ide 7.5
MD.Ide 7.0
MD.Xml 7.0 <--- note this one
MD.Core 7.5
MD.Core 7.0
MD.DesignerSupport 7.5
MD.DesignerSupport 7.0
MD.SourceEditor 7.5
MD.SourceEditor 7.0
```

So the cache ends up broken if we resolve from versioned to non-versioned.

Reply:
cc @mhutch not sure how we can get this to behave properly in all cases. Having fully versioned IDs in the cache _also_ makes it complicated to lookup in the cache dictionary.

Reply:
I didn't say we need full IDs in the cache, I said that this change can have side effects. For example:
https://github.com/mono/mono-addins/blob/bd84b756c8a113e6c0d0fc261084a846f4327b5f/Mono.Addins/Mono.Addins/ExtensionContext.cs#L1086
The check for this error depends on the exact add-in version number being provided. Using add-in names in the dependency cache is ok if the cache is refreshed when add-ins change, but let's avoid unnecessary changes in the rest of the code.